google-cloud-firestore/lib/google/cloud/firestore.rb

```diff
@@ -21,6 +21,442 @@ module Cloud
+    # databases, as a NoSQL database it differs from them in the way it
+    # describes relationships between data objects.
+    #
+    # For more information about Cloud Firestore, read the [Cloud
+    # Firestore Documentation](https://cloud.google.com/firestore/docs/).
+    #
+    # The goal of google-cloud is to provide an API that is comfortable to
+    # Rubyists. Authentication is handled by {Google::Cloud#firestore}. You can
+    # provide the project and credential information to connect to the Cloud
+    # Firestore service, or if you are running on Google Compute Engine this
+    # configuration is taken care of for you. You can read more about the
+    # options for connecting in the [Authentication
+    # Guide](https://googlecloudplatform.github.io/google-cloud-ruby/#/docs/guides/authentication).
+    #
+    # ## Adding data
+    #
+    # Cloud Firestore stores data in Documents, which are stored in Collections.
+    # Cloud Firestore creates collections and documents implicitly the first
+    # time you add data to the document. You do not need to explicitly create
+    # collections or documents. (For more information, see [Adding Data to Cloud
+    # Firestore](https://cloud.google.com/firestore/docs/manage-data/add-data).
+    #
+    # To create or overwrite a single document, use {Firestore::Client#doc} to
+    # obtain a document reference. (This does not create a document in Cloud
+    # Firestore.) Then, call {Firestore::DocumentReference#set} to create the
+    # document or overwrite an existing document:
+    #
+    # ```ruby
+    # require "google/cloud/firestore"
+    #
+    # firestore = Google::Cloud::Firestore.new
+    #
+    # # Get a document reference
+    # nyc_ref = firestore.doc "cities/NYC"
+    #
+    # nyc_ref.set({ name: "New York City" }) # Document created
+    # ```
+    #
+    # When you use this combination of `doc` and `set` to create a new document,
+    # you must specify an ID for the document. (In the example above, the ID is
+    # "NYC".) However, if you do not have a meaningful ID for the document, you
+    # may omit the ID from a call to {Firestore::CollectionReference#doc}, and
+    # Cloud Firestore will auto-generate an ID for you.
+    #
+    # ```ruby
+    # require "google/cloud/firestore"
+    #
+    # firestore = Google::Cloud::Firestore.new
+    #
+    # # Get a collection reference
+    # cities_col = firestore.col "cities"
+    #
+    # # Get a document reference with data
+    # random_ref = cities_col.doc
+    # random_ref.set({ name: "New York City" })
+    #
+    # # The document ID is randomly generated
+    # random_ref.document_id #=> "RANDOMID123XYZ"
+    # ```
+    #
+    # You can perform both of the operations shown above, auto-generating
+    # an ID and creating the document, in a single call to
+    # {Firestore::CollectionReference#add}. This operation will fail if the
+    # document already exists.
```

Comment:
"This operation will fail if the document already exists."

This should basically never happen for `add` and hence I would remove this sentence.

Reply:
Removed.

Reply:
Fixed.